charts/seed-controlplane/charts/vpn-seed-server/templates/deployment.yaml

```diff
@@ -0,0 +1,130 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: vpn-seed-server
+  namespace: {{ .Release.Namespace }}
+  labels:
+    gardener.cloud/role: controlplane
+    app: vpn-seed-server
+    networking.gardener.cloud/from-shoot-apiserver: allowed
+spec:
+  minReadySeconds: 30
+  revisionHistoryLimit: 1
+  replicas: 1
+  strategy:
+    rollingUpdate:
+      maxSurge: 100%
+      maxUnavailable: 0%
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: vpn-seed-server
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        checksum/secret-vpn-seed-server-dh: {{ include (print $.Template.BasePath "/secret-dh.yaml") . | sha256sum }}
+        checksum/secret-vpn-seed-server-tlsauth: {{ include (print $.Template.BasePath "/secret-tlsauth.yaml") . | sha256sum }}
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        gardener.cloud/role: controlplane
+        garden.sapcloud.io/role: controlplane
+        app: vpn-seed-server
+        networking.gardener.cloud/to-dns: allowed
+        networking.gardener.cloud/to-private-networks: allowed
+        networking.gardener.cloud/to-shoot-networks: allowed
+        networking.gardener.cloud/from-prometheus: allowed
+    spec:
+      priorityClassName: system-cluster-critical
+      dnsPolicy: Default # make sure to not use the coredns for DNS resolution.
```

Comment:
Why is this important here?

Reply:
`VPN-Server` and `Envoy-sidecar` do not need to resolve cluster dns names and therefore `dnsPolicy: default` is sufficient.